src/components/util/ColorSwatch.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { mostReadable } from 'tinycolor2';
+import cx from 'classnames';
+
+const CopyButton = ({ value, position, copyToClipboard }) => (
+  <button
+    onClick={() => copyToClipboard(value)}
+    className={`swatchCopyButton swatchCopyButton--${position}`}
+  >
+    {value}
+  </button>
+);
+
+CopyButton.propTypes = {
+  value: PropTypes.string,
+  position: PropTypes.string,
+  copyToClipboard: PropTypes.func,
+};
+
+const mostReadableConfig = (hexName) =>
+  mostReadable(hexName, '#333', { includeFallbackColors: true, level: 'AAA' });
+
+export const ColorSwatch = (props) => {
+  const [copiedText, setCopiedText] = useState('');
+  const [isHovered, setIsHovered] = useState(false);
+
+  const copyToClipboard = (value) => {
+    // eslint-disable-next-line no-undef
+    window.navigator.clipboard.writeText(value);
+    setCopiedText(value);
+    setTimeout(() => setCopiedText(''), 2000);
+  };
+
+  const colorName = props.jsVar.split('_')[1];
+  const hexName = `#${props.hex.toUpperCase()}`;
+  const { cssVar, jsVar } = props;
+
+  const layouts = [
+    { value: jsVar, position: 'topRight' },
+    { value: cssVar, position: 'topLeft' },
+    { value: hexName, position: 'bottomRight' },
+  ];
+
+  return (
+    <div
+      className={cx('display--inlineBlock', 'rounded--all', 'swatch-container', {
```

Comment:
probably doesn't need `inlineBlock` if it's an implicit flex child

Reply:
It may be useful in the off chance that want to render a swatch somewhere else, and don't necessarily want the logic of the wrapper? I'll remove it for now though.